### PR TITLE
Feature/preonic/rev3/alfrdmalr

### PR DESCRIPTION
**Pull request by alfrdmalr**
## Description
Created custom preonic layout (also compatible with planck).

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR
none

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/preonic/keymaps/alfrdmalr/keymap.c [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+          return false;
+          break;
+        case SETTINGS:
+          if (record->event.pressed) {
+            layer_on(_SETTINGS);
+          } else {
+            layer_off(_SETTINGS);
+          } 
+          return false;
+          break;
```
Comment (drashna) [inline keyboards/preonic/keymaps/alfrdmalr/keymap.c]: You don't need any of these. 

`MO(xxx)` will do exactly this. No need to make a custom macro for this.

File: keyboards/preonic/keymaps/alfrdmalr/keymap.c [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+ * | CTRL | C/S  | LGUI | LALT | SYMB |  SPACE/NUM  | SYMB | RALT | SETT |mu tog| LEAD |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_QWERTY] = LAYOUT_preonic_grid( \
+  KC_GRV,      KC_1,     KC_2,    KC_3,    KC_4,    KC_5,     KC_6,     KC_7,    KC_8,    KC_9,     KC_0,    KC_BSPC, \
+  KC_TAB,      KC_Q,     KC_W,    KC_E,    KC_R,    KC_T,     KC_Y,     KC_U,    KC_I,    KC_O,     KC_P,    KC_DEL,  \
+  NAVIGATION,  KC_A,     KC_S,    KC_D,    KC_F,    KC_G,     KC_H,     KC_J,    KC_K,    KC_L,     KC_SCLN, KC_QUOT, \
+  KC_LSFT,     KC_Z,     KC_X,    KC_C,    KC_V,    KC_B,     KC_N,     KC_M,    KC_COMM, KC_DOT,   KC_SLSH, KC_RSFT,  \
+  KC_LCTL,     CTRLSHFT, KC_LGUI, KC_LALT, SYMBOLS, NUMSPACE, NUMSPACE, SYMBOLS, KC_RALT, SETTINGS, MU_TOG,  KC_LEAD  \
+),
```
Comment (fauxpark) [inline keyboards/preonic/keymaps/alfrdmalr/keymap.c]: The backslashes at the end of these lines are not necessary. You can keep them if you want as it isn't hurting anything, but IMO it looks neater without them.

Comment (alfrdmalr) [reply keyboards/preonic/keymaps/alfrdmalr/keymap.c]: Thanks, looks much better without.

File: keyboards/preonic/keymaps/alfrdmalr/keymap.c (hunk 1 above)
Comment (alfrdmalr) [reply keyboards/preonic/keymaps/alfrdmalr/keymap.c]: DUH. Thanks for that - I had originally planned on using a trilayer, but decided against it and evidently never got rid of my boilerplate for it.

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**